### PR TITLE
repokitteh: Add welcome message for new contributors

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: repokitteh: Add welcome message for new contributors
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
[Optional Runtime guard:]
[Optional Fixes #Issue] Partial fix for #13232 
[Optional Deprecated:]

### CODE REVIEW

**Comment by phlax** [comment]
cc @itayd

File: ci/repokitteh/modules/newuser.star [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+def newcontributor_message(sender):
+  lines = [
+    "hi @%s, welcome and thankyou for your contribution." % sender,
+    "  We will try to review your Pull Request as quickly as possible.",
+    "  In the meantime, please take a look at the contribution guidelines if you have not done so already."]
+  github.issue_create_comment('\n'.join(lines))
+
+def _pr(action, pull_request, sender, config):
+  if action in ['opened'] and is_newcontributor(pull_request, sender):
```
Comment (phlax) [inline ci/repokitteh/modules/newuser.star:28]: s/in/=/

File: repokitteh.star [hunk 2]
```diff
@@ -5,6 +5,7 @@ use("github.com/repokitteh/modules/review.star")
 use("github.com/repokitteh/modules/wait.star")
 use("github.com/repokitteh/modules/circleci.star", secret_token=get_secret('circle_token'))
 use("github.com/envoyproxy/envoy/ci/repokitteh/modules/azure_pipelines.star", secret_token=get_secret('azp_token'))
+use("github.com/envoyproxy/envoy/ci/repokitteh/modules/newuser.star")
```
Comment (phlax) [inline repokitteh.star:8]: s/newuser/newcontributor/

File: ci/repokitteh/modules/newcontributor.star [hunk 3]
```diff
@@ -0,0 +1,28 @@
+
+# not sure if this is needed, or whether `pull_request` below has info already
+def get_pr_author_association(issue_number):
+  return github.call(
+    method="GET",
+    path="repos/envoyproxy/envoy/pulls/%s" % issue_number)["json"]["author_association"]
+
+def is_newcontributor(pull_request):
+  # check for any existing open PRs from user ?
+  return pull_request["author_association"] == "FIRST_TIME_CONTRIBUTOR"
```
Comment (phlax) [inline ci/repokitteh/modules/newcontributor.star:10]: not sure if this will work  - or whether the function above is required to get author assoc

**Comment by phlax** [comment]
The output from the welcome message can be seen here https://github.com/envoyproxy/envoy/pull/13306#issuecomment-701326650

**Comment by phlax** [comment]
@itayd i think this one should be ready for review

File: ci/repokitteh/modules/newcontributor.star [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+"""
+
+def get_pr_author_association(issue_number):
+  return github.call(
+    method="GET",
+    path="repos/envoyproxy/envoy/pulls/%s" % issue_number)["json"]["author_association"]
+
+def is_newcontributor(issue_number):
+  return get_pr_author_association(issue_number) == "FIRST_TIME_CONTRIBUTOR"
```
Comment (itayd) [inline ci/repokitteh/modules/newcontributor.star:17]: nice!

**Review by itayd — APPROVED** [review]

**Comment by phlax** [comment]
@mattklein123 it would be good to land this, and i would like to test it once it does

File: ci/repokitteh/modules/newcontributor.star [hunk 5]
```diff
@@ -0,0 +1,31 @@
+
+NEW_CONTRIBUTOR_MESSAGE = """
+hi @%s, welcome and thankyou for your contribution.
```
Comment (mattklein123) [inline ci/repokitteh/modules/newcontributor.star:3]:
```suggestion
Hi @%s, welcome and thank you for your contribution.
```

**Review by mattklein123 — CHANGES_REQUESTED** [review]
Super cool, just one small comment.

/wait

File: ci/repokitteh/modules/newcontributor.star (hunk 5 above)
Comment (mattklein123) [reply ci/repokitteh/modules/newcontributor.star]: Sorry, can you capitalize Hi?

/wait

Comment (phlax) [reply ci/repokitteh/modules/newcontributor.star]: ah, sorry, on it...

**Review by mattklein123 — APPROVED** [review]
Thanks!

**Comment by phlax** [comment]
@mattklein123 this doesnt seem to have worked on the first pass. I have pinged @itayd to see if we can trace why not

**Pull request merged**